Dockerfile.ubuntu22

```diff
@@ -0,0 +1,147 @@
+    && update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-9 10 \
+    && update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-9 10
+
+# Creating a cache dir for downloads to avoid redownloading
+ENV DOWNLOADS_CACHE_DIR=/tmp/downloads
+RUN mkdir $DOWNLOADS_CACHE_DIR
+
+# Installing Boost
+ENV BOOST_VERSION=${BOOST_VERSION_MAJOR}.${BOOST_VERSION_MINOR}.${BOOST_VERSION_PATCH}
+ENV BOOST_V_UNDERSCORE=${BOOST_VERSION_MAJOR}_${BOOST_VERSION_MINOR}_${BOOST_VERSION_PATCH}
+ENV BOOST_LABEL="boost_${BOOST_V_UNDERSCORE}"
+ENV BOOST_TARBALL="${BOOST_LABEL}.tar.gz"
+RUN --mount=type=cache,target=$DOWNLOADS_CACHE_DIR \
+    wget -N --progress=bar:force -P $DOWNLOADS_CACHE_DIR \
+        https://boostorg.jfrog.io/artifactory/main/release/${BOOST_VERSION}/source/${BOOST_TARBALL} \
+    && tar xzf $DOWNLOADS_CACHE_DIR/${BOOST_TARBALL} -C .
+# Important env variable for RonDB compilation
+ENV BOOST_ROOT="/${BOOST_LABEL}"
+
+# Ubuntu 22.04 contains openssl-3.x
+# RonDB 21.04 requires an older OpenSSL version (1.1.1m) and will look for 
+# it in $OPENSSL_ROOT. We will not overwrite the default OpenSSL, but 
+# just install the specified version
+# Commands are from https://linuxpip.org/install-openssl-linux/
+ENV OPENSSL_ROOT=/usr/local/ssl
+RUN --mount=type=cache,target=$DOWNLOADS_CACHE_DIR \
+    --mount=type=cache,target=/var/cache/apt,id=ubuntu22-apt \
+    --mount=type=cache,target=/var/lib/apt/lists,id=ubuntu22-apt-lists \
+    apt-get update -y \
+    && apt-get install -y build-essential checkinstall zlib1g-dev \
+    && wget -N --progress=bar:force -P $DOWNLOADS_CACHE_DIR \
+        https://www.openssl.org/source/openssl-$OPEN_SSL_VERSION.tar.gz \
+    && tar -xf $DOWNLOADS_CACHE_DIR/openssl-$OPEN_SSL_VERSION.tar.gz -C . \
+    && cd openssl-$OPEN_SSL_VERSION \
+    && ./config --prefix=$OPENSSL_ROOT --openssldir=$OPENSSL_ROOT shared zlib \
+    && make -j$THREADS_ARG \
+    && make install \
+    && echo "$OPENSSL_ROOT/lib" >> /etc/ld.so.conf.d/openssl-$OPEN_SSL_VERSION.conf \
+    && ldconfig -v \
+    && cd .. \
+    && rm -r openssl-$OPEN_SSL_VERSION
+    # Could also run `make test`
+    # `make install` places shared libraries into $OPENSSL_ROOT
+
+# Installing CMake
+# CMake will look for $OPENSSL_ROOT_DIR
+ENV OPENSSL_ROOT_DIR=$OPENSSL_ROOT
+RUN --mount=type=cache,target=$DOWNLOADS_CACHE_DIR \
+    wget -N --progress=bar:force -P $DOWNLOADS_CACHE_DIR \
+        https://github.com/Kitware/CMake/releases/download/v$CMAKE_VERSION/cmake-$CMAKE_VERSION.tar.gz \
+    && tar xzf $DOWNLOADS_CACHE_DIR/cmake-$CMAKE_VERSION.tar.gz -C . \
+    && cd cmake-$CMAKE_VERSION \
+    && ./bootstrap --prefix=/usr/local \
+    && make -j$THREADS_ARG \
+    && make install \
+    && cd .. \
+    && rm -r cmake-$CMAKE_VERSION
+
+RUN groupadd $USER --gid $GID && adduser $USER --uid $UID --ingroup $USER --shell /bin/bash
+
+WORKDIR /home/$USER
+
+# For the deployment of Java artifacts to the Maven repository
+COPY <<-"EOF" .m2/settings.xml
```

Comment:
How come?

Reply:
I'd like to keep this tbh, just because it becomes difficult to back-track what is missing if we want to deploy from here again for some reason. Someone else might think that file is something specific to Oraclelinux. It's going to do an early return in the deploy.sh though. I also added a comment.